app/src/main/kotlin/app/futured/androidprojecttemplate/tools/extensions/ContextExtensions.kt

```diff
@@ -0,0 +1,9 @@
+package app.futured.androidprojecttemplate.tools.extensions
+
+import android.content.Context
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.preferencesDataStore
+import app.futured.androidprojecttemplate.tools.Constants
+
+val Context.dataStore: DataStore<Preferences> by preferencesDataStore(name = Constants.DataStore.DEFAULT_DATASTORE_NAME)
```

Comment:
This extension is only used to create a datastore in Hilt module, and could be potentially misused by user to create a new datastore instead of injecting it. I would remove this and implement everything inside ApplicationModule, or at least move it to the ApplicationModule file and make it private.